Bottom: 0e0717d850658c1fbaca4673f0298caac1466776
Top:    d5fc3e852f2a881944ff9ee18c15fb8c91bff56f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-11 15:57:52 +0100

Refresh of speed-up-poly_contourincontour

---

diff --git a/src/polygon1.c b/src/polygon1.c
index e287559..f9608ca 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2396,10 +2396,8 @@ poly_ContourInContour (PLINE * poly, PLINE * inner)
   assert (inner != NULL);
   if (cntrbox_inside (inner, poly))
     {
-      /* Since computing the interior point isn't that quick,
-       * and many tests will return FALSE, try this quick test
-       * of an arbitrary external point. If it returns FALSE,
-       * the "inner" polygon is NOT inside poly.
+      /* We need to prove the "inner" contour is not outside
+       * "poly" contour. If it is outside, we can return.
        */
       if (!poly_InsideContour (poly, inner->head.point))
         return 0;
